### ISSUE TITLE
Option to Invert Y Axis

### ISSUE THREAD

**Issue by source-transformer** [cytoscape/cytoscape.js#3097]
**Description of new feature**

You can invert a canvas by providing a transform like the following:

```
context.transform(1, 0, 0, -1, 0, canvas.height)
```

While cytoscape is free to use whatever they want for rendering - it would be nice if callers could specify the origin in the rendering area.

**Motivation for new feature**

There are some platforms/systems that want the origin in the bottom left rather than top left.

**For reviewers**

_Reviewers should ensure that the following tasks are carried out for incorporated issues:_

- [ ] Ensure that the reporter has adequately described their idea.  If not, elicit more information about the use case.  You should iteratively build a spec together.
- [ ] Ensure that the issue is a good fit for the core library.  Some things are best done in extensions (e.g. UI-related features that aren't style-related).  Some things are best done by app authors themselves -- instead of in Cytoscape libraries.
- [ ] The issue has been associated with a corresponding [milestone](https://github.com/cytoscape/cytoscape.js/milestones).
- [ ] The commits have been incorporated into the `unstable` branch via pull request.  The corresponding pull request is cross-referenced.

**Comment by maxkfranz** [comment]
You're right that a transformation matrix can be applied to a canvas in order to flip it.  I don't believe that it would be pragmatic to use that particular approach as an option in the library, however.  The directions of axes are too fundamental an assumption, and many features -- and extensions -- safely assume that the axes follow the convention of canvases and of HTML generally.  For instance, dragging elements would behave strangely if that flipping transformation were applied to the canvas.

That being said, the idea of different directions for display purposes is already a feature of many (esp. directed) layouts, and layouts are what affect user-perceived direction of a network.  That's at a higher level of abstraction than the directions of the axes themselves.

For example, the grid layout has a `transform` option, which can be used to apply arbitrary transformations on the node positions that the layout outputs, similar to your canvas transformation idea:

```js
transform: function (node, position ){ return position; } // transform a given node position. Useful for changing flow direction in discrete layouts 
```
